### PR TITLE
fix: Update DecryptionMaterials code to support legacy custom CMMs

### PR DESCRIPTION
**Pull request by lucasmcdonald3**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

**Review by texastony — DISMISSED** [review]
Lucas, you nailed it.

File: src/main/java/com/amazonaws/encryptionsdk/model/DecryptionMaterials.java [hunk 1]
```diff
@@ -12,7 +13,11 @@ public final class DecryptionMaterials {
   private DecryptionMaterials(Builder b) {
     dataKey = b.getDataKey();
     trailingSignatureKey = b.getTrailingSignatureKey();
-    encryptionContext = b.getEncryptionContext();
+    if (b.getEncryptionContext() != null) {
+      encryptionContext = b.getEncryptionContext();
+    } else {
+      encryptionContext = Collections.emptyMap();
+    }
```
Comment (seebees) [inline src/main/java/com/amazonaws/encryptionsdk/model/DecryptionMaterials.java:20]: I wrote then and then though more. I don't know which one I like more. So I leave it up to the group.

```suggestion
    encryptionContext = b.getEncryptionContext();
    if (encryptionContext == null) {
      encryptionContext = Collections.emptyMap();
    }
```

File: src/test/java/com/amazonaws/crypto/examples/v2/V2DefaultCryptoMaterialsManager.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  and CMMs are expected to set this attribute.
+  The V3 commit modified this DefaultCMM's `decryptMaterials` implementation
+  to set encryptionContext on returned DecryptionMaterials objects.
+  However, there are custom implementations of the legacy native CMM
+  that do not set encryptionContext.
+  This CMM is used to explicitly assert that the V2 implementation of
+  the DefaultCMM is compatible with V3 logic,
+  which implicitly asserts that custom implementations of V2-compatible CMMs
+  are also compatible with V3 logic.
+ */
```
Comment (seebees) [inline src/test/java/com/amazonaws/crypto/examples/v2/V2DefaultCryptoMaterialsManager.java:44]: Nice comment.

**Review by seebees — APPROVED** [review]
I know you will need it again, but it looks good

File: src/main/java/com/amazonaws/encryptionsdk/model/DecryptionMaterials.java (hunk 1 above)
Comment (lucasmcdonald3) [reply src/main/java/com/amazonaws/encryptionsdk/model/DecryptionMaterials.java:20]: I like this more

Comment (lucasmcdonald3) [reply src/main/java/com/amazonaws/encryptionsdk/model/DecryptionMaterials.java:20]: Ah, never mind. encryptionContext is final. The original one has to stay.

File: src/test/java/com/amazonaws/encryptionsdk/CMMHandlerTest.java [hunk 3]
```diff
@@ -3,139 +3,158 @@
...
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.when;
+import org.junit.Test;
```
Comment (seebees) [inline src/test/java/com/amazonaws/encryptionsdk/CMMHandlerTest.java:21]: TBH, I prefer that we keep these. But I don't know how to make that work, or want to block on it.

**Review by seebees — APPROVED** [review]
LGTM

File: src/test/java/com/amazonaws/encryptionsdk/CMMHandlerTest.java (hunk 3 above)
Comment (lucasmcdonald3) [reply src/test/java/com/amazonaws/encryptionsdk/CMMHandlerTest.java:21]: We are keeping these! They just got moved to the top of the file in checkstyle reformatting

**Pull request merged**